### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by MarchD**
[DEMO LINK](https://MarchD.github.io/react_add-todo-form/)

### CODE REVIEW

**Review by olesiatk — APPROVED** [review]
Good job!

File: src/App.js [hunk 1]
```diff
@@ -1,19 +1,131 @@
...
+      errorUser: false,
+      errorTitle: false,
+    });
+
+    if (userId === 0) {
+      return this.setState({ errorUser: true });
+    }
+
+    if (taskTitle.trim().length < 1) {
+      return this.setState({ errorTitle: true });
```
Comment (vsm2k17) [inline src/App.js:47]: how did you expect this should work? why do you return the result of setState? and return it to where?

File: src/App.js [hunk 2]
```diff
@@ -1,19 +1,131 @@
...
+
+const preparedTodos = todosFromApi.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+class App extends React.Component {
+  state = {
+    todos: preparedTodos,
+    users: usersFromApi,
```
Comment (vsm2k17) [inline src/App.js:19]: I think you can avoid storing users in a state

**Review by vsm2k17 — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 3]
```diff
@@ -1,19 +1,135 @@
...
+
+    if (taskTitle.trim().length < 1) {
+      this.setState({ errorTitle: true });
+
+      return;
+    }
+
+    this.setState((state) => {
+      const newTask = {
+        id: Math.random(),
```
Comment (vsm2k17) [inline src/App.js:55]: better to use previous id + 1, think about the situation when math.random generate you the same id as you already have

**Review by vsm2k17 — APPROVED** [review]
good job

**Pull request closed without merging**